bluemira/radiation_transport/radiation_tools.py

```diff
@@ -0,0 +1,1121 @@
+    gamma: float
+        Sheath heat transmission coefficient
+    eps_cool: float
+        Electron energy loss [eV]
+    f_ion_t: float
+        Hydrogen first ionization [eV]
+    b_pol_tar: float
+        Poloidal magnetic field at the target [T]
+    b_pol_u: float
+        Poloidal magnetic field at the midplane [T]
+    alpha: float
+        Incident angle between separatrix and target plate as
+        poloidal projection [deg]
+    r_u: float
+        Upstream location radial coordinate [m]
+    r_tar: float
+        stike point radial coordinate [m]
+    lambda_q_near: float
+        Power decay length in the near SOL at the midplane [m]
+    b_tot_tar: float
+        Total magnetic field at the target [T]
+    lfs: boolean
+        low field side. Default value True.
+        If False it stands for high field side (hfs).
+
+    Returns
+    -------
+    t_tar: float
+        target temperature. Unit [keV]
+    """
+    # flux expansion at the target location
+    f_exp = (r_u * b_pol_u) / (r_tar * b_pol_tar)
+
+    # lambda target
+    # lambda_q_tar = lambda_q_near * f_exp * (1 / np.sin(np.deg2rad(alpha_pol_deg)))
+    lambda_q_tar = lambda_q_near * f_exp * (1 / np.sin(np.deg2rad(alpha_pol_deg)))
+    # wet area as poloidal section
+    a_wet = 4 * np.pi * r_tar * lambda_q_tar
+
+    # parallel cross section
+    a_par = a_wet * (b_pol_tar / b_tot_tar)
+
+    # parallel power flux density
+    q_u = p_sol / a_par
+
+    # ion mass in kg (it should be DT = 2.5*amu)
+    m_i_amu = constants.D_MOLAR_MASS
+    m_i_kg = constants.raw_uc(m_i_amu, "amu", "kg")
+
+    # converting upstream temperature
+    # upstream electron density - no fifference hfs/lfs?
+    # Numerator and denominator of the upstream forcing function
+    num_f = m_i_kg * 4 * (q_u**2)
+    den_f = 2 * E_CHARGE * (gamma**2) * (E_CHARGE**2) * (n_u**2) * (t_u**2)
+    # forcing function
+    f_ev = num_f / den_f
+
+    # Critical target temperature
+    t_crit = eps_cool / gamma
+
+    # Finding roots of the target temperature quadratic equation
+    coeff_2 = 2 * (eps_cool / gamma) - f_ev
+    coeff_3 = (eps_cool**2) / (gamma**2)
+    coeff = [1, coeff_2, coeff_3]
+    roots = np.roots(coeff)
+
+    if roots.dtype == complex:
+        t_tar = f_ion_t
+    else:
+        # Excluding unstable solution
+        sol_i = np.where(roots > t_crit)[0][0]
+
+        # Target temperature
+        t_tar = roots[sol_i]
+
+    # target temperature in keV
+    t_tar = constants.raw_uc(t_tar, "eV", "keV")
+
+    #test
+    boltz = 1.380649e-23/(8.61e-5) #[J/K]
+
+    c_st = (2 * boltz * 5/m_i_kg)**(1/2)
+
+    q_t = 7*boltz*5*7.5e20*c_st
+
+    return t_tar
+
+
+def random_point_temperature(
+    x_p: float,
+    z_p: float,
+    t_u: float,
+    p_sol: float,
+    lambda_q_near: float,
+    eq: Equilibrium,
+    r_sep_mp: float,
+    z_mp: float,
+    k_0: float,
+    firstwall_geom: Grid,
+    lfs=True,
+    connection_length=None,
+):
```

Comment:
```suggestion
    separation_correction: float,
):
```

Reply:
Cannot I just write `sep_corrector` = SeparationCorrector.DN if eq.is_double_null else SeparationCorrector.SN` instead of giving it as parameter?

Reply:
It depends whether there will ever be a situation where it needs to be specified, (eg for a custom value) if not then its not needed.

Also your above snippet wont work because the class in not initialised, if you want it to work like that it should be an enum instead